pkg/controller/migration/convert/network_test.go

```diff
@@ -527,63 +514,355 @@ var _ = Describe("Convert network tests", func() {
   }
   ]
 }`, ipam),
+					}}
+					ds.Spec.Template.Spec.Containers[0].Env = []corev1.EnvVar{{
+						Name:  "CALICO_NETWORKING_BACKEND",
+						Value: "bird",
+					}}
+					c := fake.NewFakeClientWithScheme(scheme, ds, emptyKubeControllerSpec(), pool, emptyFelixConfig())
+					_, err := Convert(ctx, c)
+					Expect(err).To(HaveOccurred())
+				},
+					Entry("subnet", `"subnet": "usePodCidr"`),
+					Entry("ipv4_pools", `"ipv4_pools": ["10.0.0.0/24"]`),
+					Entry("ipv6_pools", `"ipv6_pools": ["2001:db8::1/120"]`),
+					Entry("both pools", `"ipv4_pools": ["10.0.0.0/24"], "ipv6_pools": ["2001:db8::1/120"]`),
+				)
+			})
+		})
+
+	})
+
+	Context("with IPv6 pool", func() {
```

Comment:
Do we need a "With an IPv4 and IPv6 pool" context?

Reply:
Yep that didn't make sense either. I removed those, added a v6 pool to the top and overhauled the tests.